hs_core/views/utils.py

```diff
@@ -499,16 +510,23 @@ def rename_irods_file_or_folder_in_django(resource, src_name, tgt_name):
             for fobj in res_file_objs:
                 old_str = fobj.resource_file.name
                 new_str = old_str.replace(src_name, tgt_name)
+                logical_file = fobj.logical_file if fobj.has_logical_file else None
                 fobj.delete()
-                ResourceFile.objects.create(content_object=resource, resource_file=new_str)
+                res_file = ResourceFile.objects.create(content_object=resource,
```

Comment:
I am sure your folder rework will impact code of this PR. I will be making changes accordingly once your folder work PR gets merged.